src/handlers/ens.ts

```diff
@@ -51,14 +53,15 @@ export const fetchMetadata = async ({
   tokenId: string;
 }) => {
   try {
-    const url = `https://metadata.ens.domains/mainnet/${contractAddress}/${tokenId}`;
-    const { data } = await rainbowFetch(url, {
-      headers: {
-        Accept: 'application/json',
+    const { data } = await ensClient.query<EnsGetNameFromLabelhash>({
```

Comment:
Note: This is the recommended method to fetch an ENS name from a labelhash
https://docs.ens.domains/dapp-developer-guide/ens-as-nft#deriving-ens-name-from-tokenid